cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -962,6 +972,60 @@ aggregate_reader_metadata::get_rowgroup_metadata() const
   return rg_metadata;
 }
 
+std::unordered_map<std::string, std::vector<int64_t>>
+aggregate_reader_metadata::get_column_chunk_metadata() const
+{
+  std::unordered_map<std::string, std::vector<int64_t>> column_chunk_metadata;
+
+  std::function<void(std::string const&, int)> const populate_column_chunk_metadata =
+    [&](std::string const& path_till_now, int schema_idx) {
+      auto const& schema  = get_schema(schema_idx);
+      auto const col_path = path_till_now.empty() ? schema.name : path_till_now + "." + schema.name;
+
+      // If this is not a leaf column, keep traversing the schema tree
+      if (not schema.children_idx.empty()) {
+        for (auto const& child_idx : schema.children_idx) {
+          populate_column_chunk_metadata(col_path, child_idx);
+        }
+        return;
+      }
+
+      // Otherwise, if this is a leaf column, collect its `total_uncompressed_size` fields from
+      // all column chunks and add to the `column_chunk_metadata` map.
+      auto total_uncompressed_sizes = std::vector<int64_t>{};
+      // For each input source
+      std::for_each(thrust::counting_iterator<size_t>(0),
+                    thrust::counting_iterator(per_file_metadata.size()),
```

Comment:
I prefer `thrust::tabulate` for something like this, but it's not a blocking request.

Reply:
`thrust::tabulate` will need more bookkeeping here as we will have to keep a track of how many row groups we have processed so far.

Reply:
I don't understand
would this just be 
```
      thrust::tabulate(per_file_metadata.begin(),
                       per_file_metadata.end(),
```

Reply:
Our output is the `total_uncompressed_sizes` vector.

Reply:
my bad, forgot how limited tabulate is
carry on!